### PR TITLE
network: several cleanups for conf parsers

### PR DESCRIPTION
**Pull request by yuwata**
No functional change, just refactoring.

### CODE REVIEW

File: src/network/networkd-nexthop.c [hunk 1]
```diff
@@ -1129,66 +1129,53 @@ int manager_rtnl_process_nexthop(sd_netlink *rtnl, sd_netlink_message *message,
...
         if (section_is_invalid(nh->section))
                 return -EINVAL;
 
         if (!nh->network->manager->manage_foreign_nexthops && nh->id == 0)
-                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                         "%s: [NextHop] section without specifying Id= is not supported "
-                                         "if ManageForeignNextHops=no is set in networkd.conf. "
-                                         "Ignoring [NextHop] section from line %u.",
-                                         nh->section->filename, nh->section->line);
+                return log_nexthop_section(nh, "nexthop without specifying Id= is not supported if ManageForeignNextHops=no is set in networkd.conf.");
```
Comment (daandemeyer) [inline src/network/networkd-nexthop.c]: Capitalize nexthop now?

File: src/network/networkd-nexthop.c [hunk 2]
```diff
@@ -1146,8 +1146,15 @@ static int nexthop_section_verify(NextHop *nh) {
         if (!nh->network->manager->manage_foreign_nexthops && nh->id == 0)
                 return log_nexthop_section(nh, "nexthop without specifying Id= is not supported if ManageForeignNextHops=no is set in networkd.conf.");
 
+        if (nh->family == AF_UNSPEC)
+                nh->family = nh->gw.family;
+        else if (nh->gw.family != AF_UNSPEC && nh->gw.family != nh->family)
+                return log_nexthop_section(nh, "Family= and Gateway= settings for nexthop contradicts with each other.");
```
Comment (daandemeyer) [inline src/network/networkd-nexthop.c]: "contradict each other"

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by poettering** [comment]
lgtm, too

**Comment by yuwata** [comment]
@DaanDeMeyer and @poettering Thank you for the review. Addressed comments. Upgrading the green label.

**Pull request merged**